tests/test_agg_constraints.py

```diff
@@ -145,11 +145,49 @@ def test_aggregated_node_max_flow_same_route(model):
     assert_allclose(agg.flow, 30.0)
     assert_allclose(A.flow + B.flow, 30.0)
 
-def test_aggregated_constraint_json(model):
+def test_aggregated_constraint_json():
     model = load_model("aggregated1.json")
 
     agg = model.nodes["agg"]
     assert(agg.nodes == [model.nodes["A"], model.nodes["B"]])
     assert_allclose(agg.factors, [2.0, 4.0])
     assert_allclose(agg.max_flow, 30.0)
     assert_allclose(agg.min_flow, 5.0)
+
+def test_piecewise_constraint(model):
+    """Test using an aggregated node constraint in combination with a
+    piecewise link in order to create a minimum flow constraint of the form
+    y = mx + c, where y is the MRF, x is the upstream flow and m and c are
+    constants.
+
+    TODO: build a standard class which simplifies the construction of this setup
+
+             / -->-- X0 -->-- \
+    A -->-- Xo -->-- X1 -->-- Xi -->-- C
+             \ -->-- X2 -->-- /
+                     |
+                     Bo -->-- Bi --> D
```

Comment:
Domains.

On Mon, 11 Jul 2016, 19:37 James Tomlinson, <notifications@github.com>
wrote:

> In tests/test_agg_constraints.py
> <https://github.com/pywr/pywr/pull/174#discussion_r70312506>:
>
> >      model = load_model("aggregated1.json")
> >
> >      agg = model.nodes["agg"]
> >      assert(agg.nodes == [model.nodes["A"], model.nodes["B"]])
> >      assert_allclose(agg.factors, [2.0, 4.0])
> >      assert_allclose(agg.max_flow, 30.0)
> >      assert_allclose(agg.min_flow, 5.0)
> > +
> > +def test_piecewise_constraint(model):
> > +    """Test using an aggregated node constraint in combination with a
> > +    piecewise link in order to create a minimum flow constraint of the form
> > +    y = mx + c, where y is the MRF, x is the upstream flow and m and c are
> > +    constants.
> > +
> > +    TODO: build a standard class which simplifies the construction of this setup
> > +
> > +             / -->-- X0 -->-- \
> > +    A -->-- Xo -->-- X1 -->-- Xi -->-- C
> > +             \ -->-- X2 -->-- /
> > +                     |
> > +                     Bo -->-- Bi --> D
>
> Why can't X2 connect straight to D?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pywr/pywr/pull/174/files/fc6aa1655b38d9a59402a7efcd39130d40725f73..9f359521ae554ba7f2485ab9162d22591d51768c#r70312506>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADtDq2X3KACjJqhBNu8GKvEt7jtk8xtwks5qUo17gaJpZM4JIIzl>
> .
>